cbits/posix/find_executable.c

```diff
@@ -12,18 +12,48 @@
 #include "common.h"
 
 // the below is only necessary when we need to emulate execvpe.
-#if !defined(HAVE_execvpe)
+#if !defined(HAVE_EXECVPE)
 
-/* Return true if the given file exists and is an executable. */
-static bool is_executable(const char *path) {
-    return access(path, X_OK) == 0;
+/* A quick check for whether the given path is absolute. */
+static bool is_absolute(const char *path) {
+    return path[0] == '/';
+}
+
+static char *concat_paths(const char *path1, const char *path2) {
+    if (is_absolute(path2)) {
+        return strdup(path2);
+    } else {
+        int len = strlen(path1) + 1 + strlen(path2) + 1;
+        char *tmp = malloc(len);
+        int ret = snprintf(tmp, len, "%s/%s", path1, path2);
+        if (ret < 0) {
+            free(tmp);
+            return NULL;
+        }
+        return tmp;
+    }
+}
+
+/* Return true if the given file exists and is an executable, optionally
+ * relative to the given working directory.
+ */
+static bool is_executable(char *working_dir, const char *path) {
+    if (working_dir) {
```

Comment:
Slight potential performance improvement: check if `path` is absolute here to avoid the unnecessary `strdup`.

Reply:
Fair point. I'll implement.